lib/HttpClient/CurlClient.php

```diff
@@ -209,6 +209,13 @@ private function constructUrlAndBody($method, $absUrl, $params, $hasFile, $apiMo
                 return [$absUrl, $params];
             }
             if ('v2' === $apiMode) {
+                if (\is_array($params) && 0 === \count($params)) {
+                    // Send a request with empty body if we have no params set
+                    // Setting the second parameter as null prevents the CURLOPT_POSTFIELDS
+                    // from being set with the '[]', which is result of `json_encode([]).
+                    return [$absUrl, null];
+                }
+
```

Comment:
Is it possible to add a test to verify this?

Reply:
I tested this in dogfooding and forgot to add an actual test in this PR.